system/lib/pthread/proxying.c

```diff
@@ -85,21 +85,29 @@ typedef struct task_queue {
   int tail;
 } task_queue;
 
-static int task_queue_init(task_queue* tasks, pthread_t thread) {
+static task_queue* task_queue_create(pthread_t thread) {
+  task_queue* tasks = malloc(sizeof(task_queue));
+  if (tasks == NULL) {
+    return NULL;
+  }
   task* task_buffer = malloc(sizeof(task) * TASK_QUEUE_INITIAL_CAPACITY);
   if (task_buffer == NULL) {
-    return 0;
+    free(tasks);
+    return NULL;
   }
   *tasks = (task_queue){.thread = thread,
                         .processing = 0,
                         .tasks = task_buffer,
                         .capacity = TASK_QUEUE_INITIAL_CAPACITY,
                         .head = 0,
                         .tail = 0};
-  return 1;
+  return tasks;
 }
 
-static void task_queue_deinit(task_queue* tasks) { free(tasks->tasks); }
+static void task_queue_destroy(task_queue* tasks) {
+  free(tasks->tasks);
```

Comment:
perhaps `test_queue*` variables could be called `queue`? then this line would read `queue->tasks` which makes more sense I think.

Reply:
I'll change the name to `queue` in functions that operate primarily on `task_queue`s and keep it as `tasks` in functions that primarily operate on `em_proxying_queue`s.